src/core/ad.jl

```diff
@@ -69,8 +71,8 @@ function gradient_logp(
     ad_type = getADtype(sampler)
     if ad_type <: ForwardDiffAD
         return gradient_logp_forward(θ, vi, model, sampler)
-    else ad_type <: TrackerAD
-        return gradient_logp_reverse(θ, vi, model, sampler)
+    else
+        return gradient_logp_reverse(ad_type(), θ, vi, model, sampler)
```

Comment:
I guess a better design would be to just define `gradient_logp(backend, theta, vi, model, sampler)` instead of distinguishing between `gradient_logp_forward` and `gradient_logp_reverse`. Now that Turing supports > 2 backends IMO the branching should be handled by multiple dispatch.

BTW, I'm wondering if it would be better to replace `getADtype(sampler)` (that returns a type) with `getADbackend(sampler)` that would return an instance of an AD backend. It seems a bit strange to retrieve the type and then call `ad_type()`.

Reply:
Sounds good.

Reply:
Done